internal/backends/winit/renderer/femtovg/glcontext.rs

```diff
@@ -49,144 +51,108 @@ impl OpenGLContext {
+        window_target: &winit::event_loop::EventLoopWindowTarget<T>,
+    ) -> Result<(winit::window::Window, Self), Box<dyn std::error::Error>> {
+        let config_template_builder = glutin::config::ConfigTemplateBuilder::new();
+
+        let (window, gl_config) = glutin_winit::DisplayBuilder::new()
+            .with_preference(glutin_winit::ApiPrefence::FallbackEgl)
+            .with_window_builder(Some(window_builder.clone()))
+            .build(window_target, config_template_builder, |it| {
+                it.reduce(|accum, config| {
+                    let transparency_check = config.supports_transparency().unwrap_or(false)
+                        & !accum.supports_transparency().unwrap_or(false);
+
+                    if transparency_check || config.num_samples() < accum.num_samples() {
+                        config
+                    } else {
+                        accum
                     }
                 })
-                .unwrap();
-
-            #[cfg(target_os = "macos")]
-            if let raw_window_handle::RawWindowHandle::AppKit(
-                raw_window_handle::AppKitWindowHandle { ns_view, .. },
-            ) = _window.raw_window_handle()
-            {
-                use cocoa::appkit::NSView;
-                let view_id: cocoa::base::id = ns_view as *const _ as *mut _;
-                unsafe {
-                    NSView::setLayerContentsPlacement(view_id, cocoa::appkit::NSViewLayerContentsPlacement::NSViewLayerContentsPlacementTopLeft)
-                }
-            }
+                .expect("internal error: Could not find any matching GL configuration")
+            })?;
 
-            Self { context: not_current_gl_context.make_current(&surface).unwrap(), surface }
-        }
+        let gl_display = gl_config.display();
+
+        let raw_window_handle = window.as_ref().map(|w| w.raw_window_handle());
+
+        let gles_context_attributes = ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::Gles(Some(glutin::context::Version {
+                major: 2,
+                minor: 0,
+            })))
+            .build(raw_window_handle);
+
+        let fallback_context_attributes = ContextAttributesBuilder::new().build(raw_window_handle);
 
-        #[cfg(target_arch = "wasm32")]
+        let not_current_gl_context = unsafe {
+            gl_display
+                .create_context(&gl_config, &gles_context_attributes)
+                .or_else(|_| gl_display.create_context(&gl_config, &fallback_context_attributes))?
+        };
+
+        let window = match window {
+            Some(window) => window,
+            None => glutin_winit::finalize_window(window_target, window_builder, &gl_config)?,
+        };
+
+        let size: winit::dpi::PhysicalSize<u32> = window.inner_size();
+
+        let width: std::num::NonZeroU32 =
+            size.width.try_into().expect("new context called with zero width window");
```

Comment:
I think this shouldn't panic.  Especially since we had the case before where the size was 0.
We should either report an error, or just use 1x1 or something like that